presentation/src/main/java/bg/government/virusafe/app/registration/RegistrationFragment.kt

```diff
@@ -60,21 +64,54 @@ class RegistrationFragment :
 			}
 		}
 
-		binding.registrationTermsAndConditionsBtn.setOnClickListener {
+		binding.termsAndConditionsCheckBox.setOnCheckedChangeListener { _, isChecked ->
+			if (isChecked) {
+				setTermsAndConditionsColor(R.color.colorPrimary)
+			}
+		}
+
+		binding.registrationTermsAndConditionsTxt.setClickablePhrase(
+			fullText = "${viewModel.localizeString(I_AGREE_WITH_LBL)} " +
+					viewModel.localizeString(TERMS_N_CONDITIONS_SMALL_LBL),
+			clickablePhrase = viewModel.localizeString(TERMS_N_CONDITIONS_SMALL_LBL),
+			shouldBoldPhrase = false,
+			shouldUnderlinePhrase = true
+		) {
 			if (!canClick()) {
-				return@setOnClickListener
+				return@setClickablePhrase
 			}
-			val dialog =
-				TermsAndConditionsDialog.newInstance(!binding.termsAndConditionsCheckBox.isChecked)
-			val fm: FragmentManager = activity?.supportFragmentManager ?: return@setOnClickListener
-			dialog.setClickListener(this)
-			dialog.show(fm, TermsAndConditionsDialog::class.java.canonicalName)
+			showAgreementsDialog(
+				viewModel.localizeString(TNC_TITLE),
+				viewModel.localizeString(TNC_PART_ONE) + viewModel.localizeString(TNC_PART_TWO),
+				Agreement.TermsAndConditions,
+				!binding.termsAndConditionsCheckBox.isChecked,
```

Comment:
It is bad practice to accept that the value can be null, you should either allow passing null or pass non-nullable value here.